Survey/SurveyUnitTests/Sources/Presentation/Modules/Login/LoginViewModelSpec.swift

```diff
@@ -0,0 +1,179 @@
+//
+//  LoginViewModelSpec.swift
+//  SurveyUnitTests
+//
+//  Created by Taher on 10/11/22.
+//
+//  swiftlint:disable function_body_length closure_body_length
+
+import Combine
+import Nimble
+import Quick
+
+@testable import Survey
+
+final class LoginViewModelSpec: QuickSpec {
+
+    override func spec() {
+
+        var viewModel: LoginViewModel!
+        let baseUrl = URL(string: "https://survey-api.nimblehq.co/api/v1/oauth/token")
+        let encoder = JSONEncoder()
+
+        describe("a LoginViewModel") {
+
+            context("when a valid response is provided") {
+
+                beforeEach {
+                    guard let url = baseUrl else {
```

Comment:
- IMO your approach (or implementations) for unit testing `LoginViewModel` seems not right 🙇 The `LoginViewModel` depends on `LoginUseCaseProtocol` and `AppCoordinatorProtocol`, so all you need is creating mocked versions of them. You don't have to go down to network and repository layers.
- This PR doesn't have any unit tests for `LoginNetworkAPI`, `LoginRepository`, and `LoginUseCase`. I don't know whether it's out of this PR's scope or if you have separate tickets to do this, kindly clarify 🙏 And following up above comment, the idea to write unit tests for those layers is like this:
  - To test `LoginNetworkAPI` you will use `MockSession`.
  - To test `LoginRepository` you will create a mocked network that conforms to `NetworkAPIProtocol`
  - To test `LoginUseCase` you will create a mocked repository that conforms to `LoginRepositoryProtocol`

Reply:
I got your point.. need to make separation.. will update soon

Reply:
Agree with @Thieurom, so I think you need to create Issue to write unit test for repository and use case first before coming back to this PR

Reply:
updated in this [commit](https://github.com/Shayokh144/ic-survey-swiftui-combine/pull/45/commits/f5457957a39d6c468022d4b12d157c568b988482)